File: ELProgressHUD/ViewController.m
//
//  ViewController.m
//  ELProgressHUD
//
//  Created by 李涌辉 on 16/8/30.
//  Copyright © 2016年 LiYonghui~. All rights reserved.
//

#import "ViewController.h"
#import "ELProgressHud.h"
#import "AFNetworking.h"

@interface ViewController () <UITableViewDelegate, UITableViewDataSource>

@property (nonatomic, strong) UITableView *tableView;
@property (nonatomic, strong) NSMutableArray *dataSource;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"ELProgressHud";
    
    self.view.backgroundColor = [UIColor whiteColor];
    
    [self.view addSubview:self.tableView];
    
    
    
}

#pragma mark - tableView dataSource

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return self.dataSource.count;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return 1;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    static NSString *cellID = @"cellID";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:cellID];
    if (!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:cellID];
    }
    cell.backgroundColor = [UIColor lightGrayColor];
    cell.textLabel.text = self.dataSource[indexPath.section];
    return cell;
}

#pragma mark - tableView delegate

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    switch (indexPath.section) {
        case ELProgressHudTypeIndicator:
        {
            [ELProgressHud showHudAtView:self.view];
            [self performSelector:@selector(hide) withObject:nil afterDelay:2.0];
        }
            break;
        case ELProgressHudTypeText:
        {
            ELProgressHud *hud = [[ELProgressHud alloc] initWithHudType:ELProgressHudTypeText];
            hud.textLabel.text = @"单纯的文本框~";
            hud.delay = 1.5;
            [hud showHudAtView:self.view];
        }
            break;
        case ELProgressHudTypeProgress:
        {
            ELProgressHud *hud = [[ELProgressHud alloc] initWithHudType:ELProgressHudTypeProgress];
            [hud showHudAtView:self.view];
            AFURLSessionManager *manager = [[AFURLSessionManager alloc] initWithSessionConfiguration:nil];
            NSURLSessionDownloadTask *task = [manager downloadTaskWithRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"http://flv2.bn.netease.com/videolib3/1607/14/SrqDm6658/SD/SrqDm6658-mobile.mp4"]] progress:^(NSProgress * _Nonnull downloadProgress) {
                
                hud.el_progress = 1.0 * downloadProgress.completedUnitCount / downloadProgress.totalUnitCount;
                
            } destination:^NSURL * _Nonnull(NSURL * _Nonnull targetPath, NSURLResponse * _Nonnull response) {
                NSLog(@"%@", targetPath);
                return targetPath;
            } completionHandler:^(NSURLResponse * _Nonnull response, NSURL * _Nullable filePath, NSError * _Nullable error) {
                [ELProgressHud hideHudAtView:self.view];
            }];
            
            [task resume];
            
        }
            break;
        case ELProgressHudTypeProgressWithCancel:
        {
            ELProgressHud *hud = [[ELProgressHud alloc] initWithHudType:ELProgressHudTypeProgressWithCancel];
            [hud showHudAtView:self.view];
            AFURLSessionManager *manager = [[AFURLSessionManager alloc] initWithSessionConfiguration:nil];
            NSURLSessionDownloadTask *task = [manager downloadTaskWithRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"http://flv2.bn.netease.com/videolib3/1607/14/SrqDm6658/SD/SrqDm6658-mobile.mp4"]] progress:^(NSProgress * _Nonnull downloadProgress) {
                
                hud.el_progress = 1.0 * downloadProgress.completedUnitCount / downloadProgress.totalUnitCount;
                
            } destination:^NSURL * _Nonnull(NSURL * _Nonnull targetPath, NSURLResponse * _Nonnull response) {
                return targetPath;
            } completionHandler:^(NSURLResponse * _Nonnull response, NSURL * _Nullable filePath, NSError * _Nullable error) {
                [ELProgressHud hideHudAtView:self.view];
            }];
            
            [task resume];
            hud.elCompletion = ^() {
                [task cancel];
            };
        }
            break;
        case ELProgressHudTypeIndicatorAndText:
        {
            ELProgressHud *hud = [[ELProgressHud alloc] initWithHudType:ELProgressHudTypeIndicatorAndText];
            hud.textLabel.text = @"一盏黄黄旧旧的灯~";
            hud.delay = 2.0;
            [hud showHudAtView:self.view];
            
        }
            break;
        case ELProgressHudTypeProgressAndText:
        {
            ELProgressHud *hud = [[ELProgressHud alloc] initWithHudType:ELProgressHudTypeProgressAndText];
            hud.textLabel.text = @"静静看着凌晨黄昏~";
            [hud showHudAtView:self.view];
            AFURLSessionManager *manager = [[AFURLSessionManager alloc] initWithSessionConfiguration:nil];
            NSURLSessionDownloadTask *task = [manager downloadTaskWithRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"http://flv2.bn.netease.com/videolib3/1607/14/SrqDm6658/SD/SrqDm6658-mobile.mp4"]] progress:^(NSProgress * _Nonnull downloadProgress) {
                
                hud.el_progress = 1.0 * downloadProgress.completedUnitCount / downloadProgress.totalUnitCount;
                
            } destination:^NSURL * _Nonnull(NSURL * _Nonnull targetPath, NSURLResponse * _Nonnull response) {
                return targetPath;
            } completionHandler:^(NSURLResponse * _Nonnull response, NSURL * _Nullable filePath, NSError * _Nullable error) {
                [ELProgressHud hideHudAtView:self.view];
            }];
            
            [task resume];
        }
            break;
        case ELProgressHudTypeGif:
        {
            ELProgressHud *hud = [[ELProgressHud alloc] initWithHudType:ELProgressHudTypeGif];
            hud.el_images = @[@"11", @"12", @"13", @"14"];
            [hud showHudAtView:self.view];
            [self performSelector:@selector(hide) withObject:nil afterDelay:5.0];
        }
            break;
        case ELProgressHudTypeDrawCircle:
        {
            ELProgressHud *hud = [[ELProgressHud alloc] initWithHudType:ELProgressHudTypeDrawCircle];
            [hud showHudAtView:self.view];
         
            
            AFURLSessionManager *manager = [[AFURLSessionManager alloc] initWithSessionConfiguration:nil];
            NSURLSessionDownloadTask *task = [manager downloadTaskWithRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"http://flv2.bn.netease.com/videolib3/1607/14/SrqDm6658/SD/SrqDm6658-mobile.mp4"]] progress:^(NSProgress * _Nonnull downloadProgress) {
                
                hud.el_strokeStart = 1.0 * downloadProgress.completedUnitCount / downloadProgress.totalUnitCount;
                
            } destination:^NSURL * _Nonnull(NSURL * _Nonnull targetPath, NSURLResponse * _Nonnull response) {
                return targetPath;
            } completionHandler:^(NSURLResponse * _Nonnull response, NSURL * _Nullable filePath, NSError * _Nullable error) {
                [ELProgressHud hideHudAtView:self.view];
            }];
            
            [task resume];
        }
            break;
        default:
            break;
    }
}

- (void)hide {
    [ELProgressHud hideHudAtView:self.view];
}

#pragma mark - setter & getter

- (UITableView *)tableView {
    if (!_tableView) {
        _tableView = [[UITableView alloc] initWithFrame:self.view.bounds style:UITableViewStyleGrouped];
        _tableView.dataSource = self;
        _tableView.delegate = self;
        _tableView.rowHeight = 44;
    }
    return _tableView;
}

- (NSMutableArray *)dataSource {
    if (!_dataSource) {
        _dataSource = [NSMutableArray arrayWithArray:@[@"ELProgressHudTypeIndicator", @"ELProgressHudTypeText", @"ELProgressHudTypeProgress", @"ELProgressHudTypeProgressWithCancel", @"ELProgressHudTypeIndicatorAndText", @"ELProgressHudTypeProgressAndText", @"ELProgressHudTypeGif", @"ELProgressHudTypeDrawCircle"]];
    }
    return _dataSource;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: ELProgressHUD-swift/ViewController.swift
//
//  ViewController.swift
//  ELProgressHUD-swift
//
//  Created by 李涌辉 on 16/9/18.
//  Copyright © 2016年 Evil Lee. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {

    let cellID : String = "cellId"
    lazy var tableView : UITableView = {
      let tableView = UITableView.init(frame: self.view.bounds, style: .grouped)
        tableView.dataSource = self
        tableView.delegate = self
        tableView.rowHeight = 44
        return tableView
    }()
    let dataSource : [String] = ["ELProgressHudTypeIndicator", "ELProgressHudTypeText", "ELProgressHudTypeProgress", "ELProgressHudTypeProgressWithCancel", "ELProgressHudTypeIndicatorAndText", "ELProgressHudTypeProgressAndText", "ELProgressHudTypeGif", "ELProgressHudTypeDrawCircle"]
    
    static var float: CGFloat = 0.0
    
    // MARK: life cycle
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        self.navigationItem.title = "ELProgressHud"
        self.view.backgroundColor = UIColor.white
        self.view.addSubview(self.tableView)
        
    }

    // MARK: 隐藏hud
    @objc private func hide() {
        ELProgressHud.hideHudAtView(aView: self.view)
    }
    
    // MARK: tableView dataSource
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 1
    }
    
    func numberOfSections(in tableView: UITableView) -> Int {
        return self.dataSource.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        var cell = tableView.dequeueReusableCell(withIdentifier: cellID)
        if cell == nil {
            cell = UITableViewCell.init(style: UITableViewCellStyle.default, reuseIdentifier: cellID)
        }
        cell?.backgroundColor = UIColor.lightGray
        cell?.textLabel?.text = self.dataSource[indexPath.section]
        return cell!
    }
    
    // MARK: tableView delegate
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        switch indexPath.section {
        case ELProgressHud.ELProgressHudType.ELProgressHudTypeIndicator.hashValue:
            ELProgressHud.showHudAtView(aView: self.view)
            self.perform(#selector(hide), with: nil, afterDelay: 1.5)
            break
        case ELProgressHud.ELProgressHudType.ELProgressHudTypeText.hashValue:
            let hud : ELProgressHud = ELProgressHud.init(hudType: .ELProgressHudTypeText)
            hud.textLabel.text = "单纯的文本框~"
            hud.delay = 1.5
            hud.showHudAtView(aView: self.view)
          break
        case ELProgressHud.ELProgressHudType.ELProgressHudTypeProgress.hashValue:
            let hud: ELProgressHud = ELProgressHud.init(hudType: .ELProgressHudTypeProgress)
            hud.el_progress = 0.3
            hud.delay = 2.0
            hud.showHudAtView(aView: self.view)
            break
        case ELProgressHud.ELProgressHudType.ELProgressHudTypeProgressWithCancel.hashValue:
            let hud: ELProgressHud = ELProgressHud.init(hudType: .ELProgressHudTypeProgressWithCancel)
            hud.el_progress = 0.6
            hud.delay = 2.0
            hud.elCompletion = { ()->Void in
               NSLog("草拟吗的swift 真麻烦╮(╯▽╰)╭")
            }
            hud.showHudAtView(aView: self.view)
            break
        case ELProgressHud.ELProgressHudType.ELProgressHudTypeIndicatorAndText.hashValue:
            let hud : ELProgressHud = ELProgressHud.init(hudType: .ELProgressHudTypeIndicatorAndText)
            hud.textLabel.text = "哈萨克之火葬魔咒"
            hud.delay = 2.0
            hud.showHudAtView(aView: self.view)
            break
        case ELProgressHud.ELProgressHudType.ELProgressHudTypeProgressAndText.hashValue:
            let hud: ELProgressHud = ELProgressHud.init(hudType: .ELProgressHudTypeProgressAndText)
            hud.textLabel.text = "我打你哦~"
            hud.el_progress = 0.4
            hud.delay = 2.0
            hud.showHudAtView(aView: self.view)
            break
        case ELProgressHud.ELProgressHudType.ELProgressHudTypeGif.hashValue:
            let hud: ELProgressHud = ELProgressHud.init(hudType: .ELProgressHudTypeGif)
            hud.el_images = ["11.png", "12.png", "13.png", "14.png"]
            hud.delay = 2.0
            hud.showHudAtView(aView: self.view)
            break
        case ELProgressHud.ELProgressHudType.ELProgressHudTypeDrawCircle.hashValue:
            let hud: ELProgressHud = ELProgressHud.init(hudType: .ELProgressHudTypeDrawCircle)
            hud.el_strokeStart = 0.3
            hud.delay = 2.0
            hud.showHudAtView(aView: self.view)
            break
        default:
          break
        }
    }
    
    // MARK:
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}


